doc/GOALS.md

```diff
@@ -77,13 +76,45 @@ Elektra needs to be similar fast then current solutions.
 Ideally it should get faster because of centralized optimization
 endeavours where everyone using Elektra can benefit from.
 
-Only pay for what you need.
+> Only pay for what you need.
 
-## Non-Goals:
+# Domain-Specific Goals
+
+These goals are specific to the Elektra's domain, again in order of importance.
+
+## Suitability for Applications
+
+Elektra must be easy and robust for application (developers) to store any configuration they
+need to store. After writing configuration settings (`kdbSet`) and reading them again (`kdbGet`)
+they get the same KeySet (aka "round trip").
+
+> This means, they must be able to store keys with any name, any string or any binary data
+> as needed for their purpose.
+
+## Possibility to Represent any Configuration File Format
+
+Elektra must be powerful and flexible enough to be able to represent any configuration file
+format. We do not judge if a feature of a configuration file format is useful and support
+the development of fully-conforming parsers and emitters.
+
+> This means, that given a correctly written storage plugin, a KeySet can be found
+> that represents the content of the configuration file.
```

Comment:
Can you suggest something what you are missing here?

Reply:
I don't really know how to best formulate it, but basically the idea that:
```json
{ "a": { "b": 5, "c": 10 } }
```
Should be two keys `a/b = "5"` and `a/c = "10"` and not some other `KeySet` that contains the same information e.g. `a = "b(5) | c(10)"`.

Maybe just changing it to "[...] that represents the content _and hierarchical structure_ of the configuration file." is enough.

I know this is a bit nit-picky, but the goals should be very carefully written to avoid future confusions.

Reply:
You are totally right here, thank you for pointing out the mistake: 6a4e684492cb0318c0f0f03ce425e33de6247bfc